src/app/inventory/store/d2-item-factory.service.js

```diff
@@ -195,11 +220,19 @@ export function D2ItemFactory(
 
     const itemType = normalBucket.type || 'Unknown';
 
-    // Bungie API Bug: infusionCategoryName is missing as of CoO: https://github.com/Bungie-net/api/issues/324
-    // TODO: We should be figuring out our categories based on
-    // itemDef.itemCategoryHashes instead. Unfortunately they're
-    // localized and don't map cleanly into our existing categories.
-    const categories = itemDef.quality && itemDef.quality.infusionCategoryName ? [itemDef.quality.infusionCategoryName.replace('v300.', 'category_').toUpperCase()] : [];
+    const categories = [];
+    if (itemDef.itemCategoryHashes) {
+      for (const hash of itemDef.itemCategoryHashes) {
```

Comment:
it might be a little cleaner to use `map` here:

```js
const categories = itemDef.itemCategoryHashes ? itemDef.itemCategoryHashes.map((hash) => categoryFromHash[hash]) : [];
```

Reply:
That would leave undefineds in the list. Using a filter+map combo would work, but is less readable.

Reply:
Fair enough. `flatMap` would help you here, but it's not a biggie if it's in its own method.